### PR TITLE
TASK1: Added implementation of rock-scissor-paper game

### PR DESCRIPTION
**Pull request by MilosSomnium**

### CODE REVIEW

**Review by hrandib — COMMENTED** [review]
It is a good style to add a new line at the end of each file

File: 01_git/main.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    case 'r':
+        user_move = Rock;
+        break;
+    case 's':
+        user_move = Scissors;
+        break;
+    case 'p':
+        user_move = Paper;
+        break;
+    defualt:
```
Comment (hrandib) [inline 01_git/main.c:30]: You've declared a new label here, and this breaks the app workflow

File: 01_git/Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	$(C) rock_scissors_paper.o main.c -o rsp_game
+
+main.o: main.c
+	$(C) -c main.c
+
+rock_scissors_paper.o: rock_scissors_paper.c rock_scissors_paper.h
+	$(C) -c rock_scissors_paper.c rock_scissors_paper.h
+
+clean:
+	$(RM) *.o *.h.gch rsp_game
```
Comment (hrandib) [inline 01_git/Makefile:18]: Please add an empty new line (and for other files)

File: 01_git/rock_scissors_paper.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    if ((player == Scissors && computer == Paper) || (player == Rock && computer == Scissors) ||
+        (player == Paper && computer == Rock))
+        return Win;
+
+    return Lose;
+}
+
+enum GameValue generate_computer_move()
+{
+    return (enum GameValue) rand() % (Paper + 1);
```
Comment (hrandib) [inline 01_git/rock_scissors_paper.c:53]: It is better to declare the enum max index explicitly in the enum, as a last value

**Review by hrandib — CHANGES_REQUESTED** [review]
Please fix review comments

File: 01_git/main.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+int main()
+{
+    srand(time(0));
+    
+    printf("Please choose: rock (r) - paper (p) - scissors (s)\n");
+
+    enum GameValue computer_move = generate_computer_move();
+
+    char user_input = getchar();
+    enum GameValue user_move = Scissors;
```
[the remark this answers is not in the document]
Comment (MilosSomnium) [reply 01_git/main.c:18]: Are there any preferences, if we know that value is set below and it cannot be uninitialized?

[the remark this answers is not in the document]
Comment (hrandib) [reply 01_git/main.c:18]: It is good practice for these cases to init it to the deliberately wrong value (enum could be extended). In this case you the mistyping of 'default' will be revealed in the earlier step.

**Review by hrandib — APPROVED** [review]

**Pull request merged**